src/providers/ogr/qgsogrexpressioncompiler.cpp

```diff
@@ -43,6 +43,8 @@ QgsSqlExpressionCompiler::Result QgsOgrExpressionCompiler::compile( const QgsExp
     return Fail;
   else if ( mSource->mDriverName == "MSSQLSpatial" )
     return Fail;
+  else if ( mSource->mDriverName == "GPKG" )
```

Comment:
I wasn't sure if that line was needed, but I added this since SQLite is listed and GeoPackage is sqlite based. I'll have a look at the spatialite expression compiler.